src/code_800AF9B0.c

```diff
@@ -3401,12 +3303,9 @@ void func_800B5404(s32 arg0, s32 arg1) {
 GLOBAL_ASM("asm/non_matchings/code_800AF9B0/func_800B5404.s")
 #endif
 
-u8 func_800B54EC(s32, s32);
-u8 func_800B5508(s32, s32, s32);
-
 // Get Grand Prix points for a given cup and CC mode
 u8 func_800B54C0(s32 cup, s32 cc_mode) {
-    return func_800B54EC(cup, D_8018ED10[cc_mode]);
+    return func_800B54EC(cup, *(&D_8018ED10 + cc_mode));
```

Comment:
D_8018ED10 is definitely a weird variable. Here its clearly meant to be used like its the head of an array, but almost everywhere else its just a single u8 for the 50 CC Grand Prix points. I don't know if there's a clean way to have it do both those things

Reply:
Yeah, I went back and forth between leaving it as an array or a u8. I thought it was a bit nicer as a u8 for consistency since there's a few places where D_8018ED11, D_8018ED12, and D_8018ED13 are referenced directly, but I don't feel too strongly either way.

Reply:
I toyed around with the idea of leaving it as a u8 and then doing ```#define allGpPoints (&D_8018ED10)``` but that's pretty ugly

Reply:
Maybe it's a union?
There's a hack you can do to have struct and array access using a union. But I suppose that doesn't quite fit this scenario...

Reply:
I'll try a union and see what you guys think